content/projects/gsoc/2021/projects/git-credentials-binding.adoc

```diff
@@ -50,35 +50,90 @@ They will be able to use the git command line options of their choice, including
 Users will be able to run authenticated git commands in their Jenkins Pipelines without modifying the git plugin.
 
 === Implementation
-Both `gitUsernamePassword` and `gitSshPrivateKey` bindings use the https://plugins.jenkins.io/credentials/[Credential Plugin]
-to retrieve user's credential using the Credentials API.
-The https://plugins.jenkins.io/credentials-binding/[Credentials Binding Plugin]
-is used to bind Git specific environment variables with programs to perform authentication on behalf of the user, without their interaction with the command-line.
+Both `gitUsernamePassword` and `gitSshPrivateKey` bindings depends on the https://plugins.jenkins.io/credentials/[Credential Plugin]
+to retrieve user's credential using the Credentials API and https://plugins.jenkins.io/credentials-binding/[Credentials Binding Plugin]
+to bind Git specific environment variables with script(sh,bat,powershell) / terminal commands depending upon the current CLI-git version installed on
+system which is being used to perform git operation that requires authentication.
```

Comment:
This is a very long sentence that should be split into multiple sentences.

We prefer sentence per line in files like this.  When we write with a sentence per line, it is easier to review and comment and is easier to see when a sentence is too long.